### ISSUE TITLE
Support Japanese wordlist for BIP39

### ISSUE THREAD

**Issue by dabura667** [hivewallet/hive-js#218]
Finally merged.

https://github.com/bitcoin/bips/pull/92/files

Japanese language generates Japanese seeds?

Judging from the NFKD issues discovered the other week, maybe it's a little too early... (certain browsers might have varying support for `.normalize('NFKD')`...

Any thoughts?

**Comment by w-hive** [comment]
@mattatgit This is another one that I would like you to look at and consider (in a general sense) for Hive 2.0.

@dabura667 how many other complete BIP39 word lists are out there for different languages now? Is anyone else using them?

**Comment by mattatgit** [comment]
@w yep. My take: only a very small minority of users actually memorise their seed phrases, in which case having a native language version doesn't help most users. Additionally, Saori's take is that a system that requires her to write down Japanese words rather than English for a password-type system is actually odder than using English words. Perhaps for people who don't read any English at all, but these people would be very unlikely to be using a bitcoin wallet in the first place (virtually all Japanese of the demographic of Hive users should be able to at least read English if not speak it so well)

**Comment by dabura667** [comment]
Among hard-core bitcoiners in Tokyo the most common "omg I goofed" story I hear is misspelling the english words from the phrase they wrote down.

Among newbies who use breadwallet (currently biggest bitcoin wallet in Japan) the biggest complaint we get to support is "make writing down the phrase easier" when asked if writing down simple hiragana words would be easier, all respondents said yes.

Sure, if you let the user choose their own password, most Japanese are used to writing down roman ccharacter password. But it is almost always a romanizing of Japanese words.

I asked a few people if their commonly used password was an english word and everyone said no. Most said it was a japanese word romanized.

English native speakers can look at a word with one glance and write it down without looking up after every letter, but I have seen even veteran bitcoiners in Tokyo who speak great English have to always look back after every letter when writing down.

breadwallet next release will support Japanese BIP39 btw.

![image](https://cloud.githubusercontent.com/assets/6586275/8718439/6242b086-2bdf-11e5-9ea8-8215e77b21ad.jpg)

**Comment by w-hive** [comment]
In your face, @mattatgit :)

**Comment by mattatgit** [comment]
@w smartass. Sure, i would have guessed they would have stored them in good password managers, (in which case, who cares what the seed phrase is, right ?) but i'm not going to argue against it if there is solid evidence for it (if anecdotal). I'd never write mine down because then i'd never be able to access it when i actually need it.

**Comment by dabura667** [comment]
Hmm, perhaps it depends on the method you wish to store the phrase in.

With breadwallet, we have warnings everywhere saying don't use password managers and don't use any device to write down the phrase. And encourage the user to write it down physically on paper.

However, I don't think there is a wrong or right, really. Just let the market decide.

I would be interested to see a romanized Japanese language wordlist though...

**Comment by dabura667** [comment]
@w-hive also I don't think anyone is using non-English BIP39 yet, as everyone only cares about bitcoiners.

Breadwallet asked to be user friendly to grandma, grandpa, and even young kids, and we decided to put in all languages based on Language setting of the device.

Also, one note on implementation.

1. Since Japanese is a language where text wraps after any letter, breadwallet had problems with some words looking like two separate words due to text wrap. Some plan must be made to ensure line break does not cut into a word when showing to the user.

**Comment by mattatgit** [comment]
@dabura667 yeah i think this also depends on what the perceived usage for a mobile wallet is. To me, that means something you probably don't carry that much money around in, like a physical wallet most of the time. In which case being secure is important, but being handy is also important. If we make bitcoin less usable and more of a hassle than plain ole cash, why bother using it on your mobile ? That's why i think encouraging password managers is better because it means that users can actually access their wallets when they need to rather than realising their piece of paper is at home in their desk drawer & abandoning the app. 

All that aside, i think we could use a JP wordlist & see how it goes. And yep, aware of the word wrap problem, thanks for the reminder!
